test/jdk/javax/swing/JComboBox/ComboPopupBug.java

```diff
@@ -21,66 +21,72 @@
+ * @run main ComboPopupBug
  */
 
 public class ComboPopupBug {
-    private static final String instructionsText = """
-            This test is used to verify that clicking on JComboBox
-            when frame containing it is about to close should not
-            cause IllegalStateException.
-
-            A JComboBox is shown with Close button at the bottom.
-            Click on Close and then click on JComboBox arrow button
-            to try to show combobox popup.
-            If IllegalStateException is thrown, test will automatically Fail
-            otherwise click Pass.""";
+    private static JFrame frame;
+    private static JButton closeButton;
+    private static JComboBox<String> comboBox;
+    private static Point comboBoxLocation;
+    private static Dimension comboBoxSize;
+    private static JRobot robot;
 
     public static void main(String[] args) throws Exception {
-        PassFailJFrame.builder()
-                .title("ComboPopup Instructions")
-                .instructions(instructionsText)
-                .testTimeOut(5)
-                .rows(10)
-                .columns(35)
-                .testUI(ComboPopupBug::createUI)
-                .build()
-                .awaitAndCheck();
-    }
+        robot = JRobot.getRobot();
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("ComboPopup");
 
-    private static JFrame createUI() {
-        JFrame frame = new JFrame("ComboPopup");
+            comboBox = new JComboBox<>();
+            comboBox.setEditable(true);
+            comboBox.addItem("test");
+            comboBox.addItem("test2");
+            comboBox.addItem("test3");
 
-        JComboBox<String> cb = new JComboBox<>();
-        cb.setEditable(true);
-        cb.addItem("test");
-        cb.addItem("test2");
-        cb.addItem("test3");
+            closeButton = new JButton("Close");
+            closeButton.addActionListener((e) -> {
+                        clickComboBox();
+                        frame.setVisible(false);
+                    });
 
-        JButton b = new JButton("Close");
-        b.addActionListener(
-                (e)->{
-                    try {
-                        Thread.sleep(3000);
-                    } catch (Exception ignored) {
-                    }
-                    frame.setVisible(false);
-                });
+            frame.getContentPane().add(comboBox, "North");
+            frame.getContentPane().add(closeButton, "South");
+            frame.setSize(200, 200);
+            frame.setVisible(true);
+        });
 
-        frame.getContentPane().add(cb, "North");
-        frame.getContentPane().add(b, "South");
-        frame.setSize(200, 200);
+        comboBoxLocation = comboBox.getLocationOnScreen();
+        comboBoxSize = comboBox.getSize();
+
+        try {
+            SwingUtilities.invokeAndWait(() -> closeButton.doClick());
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Swing is not thread-safe, you should get the location and size on EDT too.
```suggestion
        try {
            SwingUtilities.invokeAndWait(() ->{
                comboBoxLocation = comboBox.getLocationOnScreen();
                comboBoxSize = comboBox.getSize();
                
                closeButton.doClick();
            });
        }
        catch (Exception e) {
            throw new RuntimeException(e);
```

Reply:
By the way, since the `closeButton` is clicked on EDT, as it should, you can put both `comboBoxLocation` and `comboBoxSize` into `clickComboBox()` where they're used.